components/raftstore-v2/src/operation/ready/mod.rs

```diff
@@ -934,6 +934,25 @@ impl<EK: KvEngine, ER: RaftEngine> Storage<EK, ER> {
         let prev_raft_state = self.entry_storage().raft_state().clone();
         let ever_persisted = self.ever_persisted();
 
+        // If snapshot initializes the peer, we don't need to write apply trace again.
+        if !ever_persisted {
```

Comment:
The comment above line explains `self.ever_persisted()` is used.

Reply:
Oh, so you mean this block should be placed after `apply_snapshot`. It's really not that obvious especially when `ever_persisted` is cached and used later in the function.